### DISCUSSION TITLE
MPR121 support as a Capacitive touch, IO expander and PWM LED driver

### DISCUSSION THREAD

**Discussion by alextrical** [davetcc/IoAbstraction#184] [Ideas]
[MPR121 GPIO and LED Driver Function](https://www.nxp.com/docs/en/application-note/AN3894.pdf)
I was wondering if it would be possible to get support for the MPR121 added to your library, as all the current implementations on Arduino only cover a fraction of the functionality of the device.
The chip has 12 Capacitive touch inputs, 8 of which can be used as GPIO or PWM generation for driving LED's

I would be happy to provide you with hardware (and/or donation) for testing if that would be of any assistance, as I believe this would be an amazing chip for the price for all the functionality it has to offer.

Once again thank you for your amazing work.

**Comment by davetcc** [comment]
Thanks for pointing out this chip, it sounds very interesting. I'll take a look at how feasible it is once I've finished the 3.0 stability work.

**Comment by davetcc** [comment]
Looks fairly straightforward to implement, Given how cheap they are, I just ordered a break-out board for testing. Once it arrives I'll have a go at it

**Comment by davetcc** [comment]
See https://github.com/davetcc/IoAbstraction/issues/185
